### PR TITLE
Testing2

### PR DESCRIPTION
**Pull request by tejashrighogare**

### CODE REVIEW

File: Assignment1/Occurancesassign.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package Occurances;
+public class Occurancesassign {
+	public static void main(String args[]) {
+		String input = "abacbc";
+		char search = 'a'; // Character to search is 'a'.
+		char search1 = 'b'; // Character to search is 'b'.
+		char search2 = 'c'; // Character to search is 'c'.
+		int count = 0;
```
Comment (amitsahu8) [inline Assignment1/Occurancesassign.java]: Logic is not proper. Write the core logic in different method and parameterize it. Also logic should be generic.

File: Assignment2/Assignment2.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class Assignment2{
+ public static String destinationCity(List<List<String>> paths)
+ {
+ List<String> out=new LinkedList<>();
+ List<String> in=new LinkedList<>();
+for(int i=0; i<paths.size();i++)
+{
+List<String>Elements=paths.get(i);
+String outElement=Elements.get(0);
+String inElement=Elements.get(1);
```
Comment (amitsahu8) [inline Assignment2/Assignment2.java]: code format issue

File: Assignment2/Assignment2.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	 	if(!out.contains(inElement))
+	 	{
+	 		return inElement;
+	 	}
+ 	}
+ return null;
+ }
+ public static void main(String args[])
+ {
+	 List<List<String>> testList=new LinkedList<>();
```
Comment (amitsahu8) [inline Assignment2/Assignment2.java]: need to know the reason why you used LinkedList

File: Assignment2/Assignment2.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.LinkedList;
+import java.util.List;
+public class Assignment2{
+ public static String destinationCity(List<List<String>> paths)
+ {
+ List<String> out=new LinkedList<>();
+ List<String> in=new LinkedList<>();
+for(int i=0; i<paths.size();i++)
+{
+List<String>Elements=paths.get(i);
```
Comment (amitsahu8) [inline Assignment2/Assignment2.java]: Variables should be camelcase

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: Assignment1/Occurancesassign.java (hunk 1 above)
Comment (tejashrighogare) [reply Assignment1/Occurancesassign.java]: Used parameterized method

File: Assignment2/Assignment2.java (hunk 2 above)
Comment (tejashrighogare) [reply Assignment2/Assignment2.java]: Done correction

File: Assignment2/Assignment2.java (hunk 4 above)
Comment (tejashrighogare) [reply Assignment2/Assignment2.java]: Done

File: Assignment2/Assignment2.java (hunk 3 above)
Comment (tejashrighogare) [reply Assignment2/Assignment2.java]: Used ArrayList.

File: mediumDifficultyAll Pro/Medium difficulty Program 3/swapPrimeNo/swapPrimeNo.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package swapPrimeNo;
+
+public class swapPrimeNo {
+    /*
+    Q3. Given an array of ints, swap the first and last elements in the array. Return the modified array.
+     If the last array value is prime no.
+     */
+    public static void main(String[] args) {
+        int[] inputArray = {1, 2, 3, 7};
+        SwapPrimeNumber(inputArray);
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 3/swapPrimeNo/swapPrimeNo.java]: Use Multiple Input parameter.

File: mediumDifficultyAll Pro/Medium difficulty Program 3/swapPrimeNo/swapPrimeNo.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        int lastElement = inputArray[inputArray.length - 1];
+            if (lastElement % 2 != 0) {
+                int temporary = 0;
+                temporary = inputArray[0];
+                inputArray[0] = inputArray[inputArray.length - 1];
+                inputArray[inputArray.length - 1] = temporary;
+            }
+            for (int i = 0; i < inputArray.length; i++)
+                System.out.print(inputArray[i] + " ");
+        }
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 3/swapPrimeNo/swapPrimeNo.java]: No need to use loop for printing Array. Use  System.out.println(Arrays.toString(inputArray));

File: mediumDifficultyAll Pro/Medium difficulty Program 7/ascendingOrderArray/ascendingOrderArray.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        int j = 0;//for next element
+        for (int i = 0; i < arrayElement - 1; i++) {
+            if (inputArrayElement[i] != inputArrayElement[i + 1]) {
+                inputArrayElement[j++] = inputArrayElement[i];
+            }
+        }
+        inputArrayElement[j++] = inputArrayElement[arrayElement - 1];
+        for (int i = 0; i < j; i++) {
+            System.out.println(inputArrayElement[i]);
+        }
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 7/ascendingOrderArray/ascendingOrderArray.java]: no need to print loop for prinit array

File: mediumDifficultyAll Pro/Medium difficulty Program 4/stringMergeAlternativly/stringMergeAlternativly.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package stringMergeAlternativly;
+public class stringMergeAlternativly {
+    /*
+    Q4. Given two strings, a and b, create a bigger string made of the first char of a, the first char of b,
+     the second char of a, the second char of b, and so on. Any leftover chars go at the end of the result.
+ */
+    public static void main(String args[])
+    {
+        String result=   mergeStringAlternatively("abc", "xyz");
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 4/stringMergeAlternativly/stringMergeAlternativly.java]: format the code using ctrl+alt+L

File: mediumDifficultyAll Pro/Medium difficulty Program 2/Patterns/Patterns.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+            for (int j = 1; j <= rows - i; j++) {
+                System.out.print(" ");
+            }
+            for (int k = i; k >= 1; k--) {
+                System.out.print(k);
+            }
+            for (int l = 2; l <= i; l++) {
+                System.out.print(l);
+            }
+            System.out.println();
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 2/Patterns/Patterns.java]: why empty syout?

File: mediumDifficultyAll Pro/Medium difficulty Program 1/taxSalary/taxSalary.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+            case 2: {
+                tax = (0.15 * (salary - 700000)) + (0.1 * 200000);
+                break;
+            }
+            case 3: {
+                tax = (0.2 * (salary - 1000000)) + (0.15 * 300000) + (0.1 * 200000);
+                break;
+            }
+
+            default: {
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 1/taxSalary/taxSalary.java]: remove extra space

File: mediumDifficultyAll Pro/Medium difficulty Program 1/taxSalary/taxSalary.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    Rs. 10,00,000-15,00,000 20% on additional amount
+
+    Above 15,00,000 30% on additional amount
+    Result: - income tax is …………………………….
+    */
+    public static void main(String[] args) {
+        taxSalaryCount(1000000);
+    }
+    public static void taxSalaryCount(double salary) {
+        double tax = 0;
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 1/taxSalary/taxSalary.java]: why using double? Practically salary will be in int. If the  calculated tax is in double format use rounding off.

File: javaTutorialPrograms/java/Occurances/funOverloading.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        return (firstNumber + secondNumber + thirdNumber);
+    }
+    public double Add(double firstNumber, double secondNumber) {
+        return (firstNumber + secondNumber);
+    }
+}
+
+
+
+
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/funOverloading.java]: remove extra space.

File: javaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+            }
+            if (firstOccCount != tempOccCount) {
+                result = false;
+                return result;
+            }
+        }
+        return result;
+    }
+}
+
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java]: remove extra space.

File: javaTutorialPrograms/java/Occurances/Hashset1.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
+
+package Occurances;
+import java.util.*;
+public class Hashset1 {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/Hashset1.java]: Change Class name do not have Integer in it

File: javaTutorialPrograms/java/Occurances/Hashset1.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+		hashsetObject.add("Three");
+		hashsetObject.add("One");
+		hashsetObject.add("Seven");
+		hashsetObject.add(null);
+		Iterator<String> i = hashsetObject.iterator();
+		while (i.hasNext()) {
+			System.out.println(i.next());
+		}
+	}
+
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/Hashset1.java]: remove extra space

File: javaTutorialPrograms/java/Occurances/abstractClass.java [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+    void run() {
+        System.out.println("Running safely");
+    }
+    //Write sample program for abstract class
+    public static void main(String args[]) {
+        simpleClass object = new simpleClass();
+        object.run();
+    }
+}
+
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/abstractClass.java]: remove extra space

File: javaTutorialPrograms/java/Occurances/arrayJava.java [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    public static void arrayElement(int elementsValue[]) {
+// Print the array elements
+        for (int i = 0; i < elementsValue.length; i++)
+            System.out.println(elementsValue[i]);
+    }
+}
+
+
+
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/arrayJava.java]: remove extra space.

File: javaTutorialPrograms/java/Occurances/arrayList_linkedList.java [hunk 18]
```diff
@@ -0,0 +1,19 @@
+package Occurances;
+import java.util.*;
+public class arrayList_linkedList {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/arrayList_linkedList.java]: Class name first letter should be capital.

File: javaTutorialPrograms/java/Occurances/dataTypes.java [hunk 19]
```diff
@@ -0,0 +1,15 @@
+package Occurances;
+public class dataTypes {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/dataTypes.java]: class first letter should be capital

File: javaTutorialPrograms/java/Occurances/default_parameterized_constructor.java [hunk 20]
```diff
@@ -0,0 +1,15 @@
+package Occurances;
+class constructorClass {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/default_parameterized_constructor.java]: Class first letter should be capital

File: javaTutorialPrograms/java/Occurances/arrayList_linkedList.java (hunk 18 above)
Comment (rohitlamsonge66) [reply javaTutorialPrograms/java/Occurances/arrayList_linkedList.java]: Do not use Under Score in it Use camel case.

File: javaTutorialPrograms/java/Occurances/equalsHashcode.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
+//Write program for equals and hashcode methods
+package Occurances;
+class Test_hash_equal {
+    public static void main(String[] args) {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/equalsHashcode.java]: Please Remove Under score from class name.

File: javaTutorialPrograms/java/Occurances/findDestination.java [hunk 22]
```diff
@@ -0,0 +1,44 @@
+//Write program for finding distination city
+package findDestination;
+import java.util.ArrayList;
+import java.util.List;
+public class findDestination {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/findDestination.java]: First letter should be capital. in class name

File: javaTutorialPrograms/java/Occurances/forEach.java [hunk 23]
```diff
@@ -0,0 +1,13 @@
+
+package Occurances;
+public class forEach {
+    //Write program for foreach loop
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/forEach.java]: First letter should be capital. in class name

File: javaTutorialPrograms/java/Occurances/funOverloading.java [hunk 24]
```diff
@@ -0,0 +1,19 @@
+package Occurances;
+public class funOverloading {
+    //Program for functionOverloading
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/funOverloading.java]: First letter should be capital. in class name

File: javaTutorialPrograms/java/Occurances/funOverriding.java [hunk 25]
```diff
@@ -0,0 +1,13 @@
+package Occurances;
+public class funOverriding {
+    void run() {
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/funOverriding.java]: First letter should be capital. in class name

File: javaTutorialPrograms/java/Occurances/hashMap.java [hunk 26]
```diff
@@ -0,0 +1,16 @@
+package Occurances;
+import java.util.*;
+public class hashMap {
+	//Write program for hashMap
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/hashMap.java]: First letter should be capital. in class name

File: javaTutorialPrograms/java/Occurances/hierarchicalInheritance.java [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+        System.out.println("barking...");
+    }
+}
+class Cat extends Animal {
+    void meow() {
+        System.out.println("meowing...");
+    }
+}
+class hierarchicalInheritance {
+    //Write program for hirarchical Inheritance
```
Comment (rohitlamsonge66) [inline javaTutorialPrograms/java/Occurances/hierarchicalInheritance.java]: First letter should be capital. in class name

File: mediumDifficultyAll Pro/Medium difficulty Program 7/ascendingOrderArray/ascendingOrderArray.java [hunk 28]
```diff
@@ -0,0 +1,37 @@
+package ascendingOrderArray;
+import java.util.*;
+public class ascendingOrderArray {
```
Comment (rohitlamsonge66) [inline mediumDifficultyAll Pro/Medium difficulty Program 7/ascendingOrderArray/ascendingOrderArray.java]: Changes for all the class name.

File: JavaTutorialPrograms/java/Occurances/InterfaceJava.java [hunk 29]
```diff
@@ -0,0 +1,16 @@
+package Occurances;
+
+interface interfaceClass {
```
Comment (rohitlamsonge66) [inline JavaTutorialPrograms/java/Occurances/InterfaceJava.java]: First letter Capital of interface name

**Review by rohitlamsonge66 — APPROVED** [review]

File: JavaTutorialPrograms/java/Occurances/AbstractClass.java [hunk 30]
```diff
@@ -0,0 +1,14 @@
+package Occurances;
+abstract class AbstarctClass {
```
Comment (amitsahu8) [inline JavaTutorialPrograms/java/Occurances/AbstractClass.java]: public missing

File: JavaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java [hunk 31]
```diff
@@ -0,0 +1,34 @@
+package Occurances;
+public class CountOccurrenceOfChar {
+    //Write a prgram to find the character occurances of given string
+    public static void main(String[] args) {
+        boolean result = sameOccurrence_new("abcdabcd");
```
Comment (amitsahu8) [inline JavaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java]: why sameOccurrence_new?
I already told not to use test, new, demo string

File: JavaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+        if (result == true)
+            System.out.println("All characters having equal number of occurances");
+        else
+            System.out.println("All characters not having equal number of occurances");
+    }
+    public static boolean sameOccurrence_new(String inputString) {
+        boolean result = true;
+        int firstOccCount = 1;
+        for (int counter = 0; counter < inputString.length(); counter++) {
+            char c = inputString.charAt(counter);
```
Comment (amitsahu8) [inline JavaTutorialPrograms/java/Occurances/CountOccurrenceOfChar.java]: why c?

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: JavaTutorialPrograms/java/findDestination/FindDestination.java [hunk 33]
```diff
@@ -0,0 +1,41 @@
...
+        listItemFirst.add("New York");
+        List<String> listItemSecond = new ArrayList<>();
+        listItemSecond.add("New York");
+        listItemSecond.add("Lima");
+        List<String> listItemThird = new ArrayList<>();
+        listItemThird.add("Lima");
+        listItemThird.add("Sao Paulo");
+        cityList.add(listItemFirst);
+        cityList.add(listItemSecond);
+        cityList.add(listItemThird);
```
Comment (prashantkawa) [inline JavaTutorialPrograms/java/findDestination/FindDestination.java]: Use better approach to add values in the list

File: JavaTutorialPrograms/java/findDestination/FindDestination.java [hunk 34]
```diff
@@ -0,0 +1,41 @@
...
+        List<String> outElemetsList = new ArrayList<>();
+        List<String> inElementsList = new ArrayList<>();
+        for (List<String> list : listInput) {
+            String outElement = list.get(0);
+            String inElement = list.get(1);
+            outElemetsList.add(outElement);
+            inElementsList.add(inElement);
+        }
+        for (String stringElement : inElementsList) {
+            String inElement = stringElement;
```
Comment (prashantkawa) [inline JavaTutorialPrograms/java/findDestination/FindDestination.java]: no need of creating this variable

File: JavaTutorialPrograms/java/queueExample/Queue.java [hunk 35]
```diff
@@ -0,0 +1,50 @@
+package queueExample;
+
+public class Queue {
+    public static int front, rear, capacity;
+    public static int queue[];
```
Comment (prashantkawa) [inline JavaTutorialPrograms/java/queueExample/Queue.java]: use private here

File: JavaTutorialPrograms/java/sortNumber/SortNumber.java [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+        printArray(array);
+        sortArray(array);
+    }
+
+    static int length;
+
+    public static void printArray(int[] array) {
+        for (int counter = 0; counter < length; counter++) {
+            System.out.print(array[counter] + " ");
+        }
```
Comment (prashantkawa) [inline JavaTutorialPrograms/java/sortNumber/SortNumber.java]: No need to use for loop to print Array values, you can use Arrays.toString() method

File: JavaTutorialPrograms/java/staticVaribleExample/Student.java [hunk 37]
```diff
@@ -0,0 +1,16 @@
+package staticVaribleExample;
+
+public class Student {
+    int rollno;
+    String name;
+    static String staticVariable = "ITS";
```
Comment (prashantkawa) [inline JavaTutorialPrograms/java/staticVaribleExample/Student.java]: use private here

File: Medium Difficulty/Medium difficulty program 1/taxPackage/TaxSalary.java [hunk 38]
```diff
@@ -0,0 +1,43 @@
...
+                taxAmount = (0.15 * (salary - 700000)) + (0.1 * 200000);
+                break;
+            }
+            case 3: {
+                taxAmount = (0.2 * (salary - 1000000)) + (0.15 * 300000) + (0.1 * 200000);
+                break;
+            }
+            default: {
+                taxAmount = (0.3 * (salary - 1500000)) + (0.2 * 500000) + (0.15 * 300000) + (0.1 * 200000);
+                break;
```
Comment (prashantkawa) [inline Medium Difficulty/Medium difficulty program 1/taxPackage/TaxSalary.java]: write this logic in case 4 and use default for invalid input

**Review by prashantkawa — CHANGES_REQUESTED** [review]

**Review by prashantkawa — APPROVED** [review]
Approving PR

**Review by prashantkawa — APPROVED** [review]
Approving PR

File: swapPrimeNo/SwapPrimeNo.java [hunk 39]
```diff
@@ -5,41 +5,47 @@
 public class SwapPrimeNo {
     private int temporary;
     private int lastNumber;
-   private boolean flag = false;
+    private boolean flag = false;
```
Comment (prashantkawa) [inline swapPrimeNo/SwapPrimeNo.java]: Remove unused variables

File: swapPrimeNo/SwapPrimeNo.java [hunk 40]
```diff
@@ -5,41 +5,47 @@
...
-    public void swapPrimeNumber(int inputArray[]) {
-        lastNumber = inputArray[inputArray.length - 1];
-        int midNumber = lastNumber / 2;
-        for (int counter = 2; counter <= midNumber; counter++) {
-            if (lastNumber % counter == 0) {
-                flag = true;
-                break;
-            }
+    public static boolean swapPrimeNumber(int number, int counter) {
+        boolean result = false;
```
Comment (prashantkawa) [inline swapPrimeNo/SwapPrimeNo.java]: Same here

File: swapPrimeNo/SwapPrimeNo.java [hunk 41]
```diff
@@ -5,41 +5,47 @@
...
     public static void main(String[] args) {
-        SwapPrimeNo swapObject = new SwapPrimeNo();
         int[] inputArray = {1, 2, 3, 7};
         int[] inputArraySecond = {1, 2, 3};
         int[] inputArrayThird = {1, 6, 7, 9, 8};
-        swapObject.swapPrimeNumber(inputArray);
-        swapObject.swapPrimeNumber(inputArraySecond);
-        swapObject.swapPrimeNumber(inputArrayThird);
+        int lastNumber = inputArray[inputArray.length - 1];
+        swapPrimeNumber(lastNumber, 2);
```
Comment (prashantkawa) [inline swapPrimeNo/SwapPrimeNo.java]: No need to call this method here as it is already getting called in printResult method

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: Medium Difficulty/Medium difficulty program 3/SwapPrimeNo/SwapPrimeNo.java [hunk 42]
```diff
@@ -5,41 +5,42 @@
 public class SwapPrimeNo {
     private int temporary;
     private int lastNumber;
```
Comment (prashantkawa) [inline Medium Difficulty/Medium difficulty program 3/SwapPrimeNo/SwapPrimeNo.java]: please remove this variables if not in use

**Review by prashantkawa — CHANGES_REQUESTED** [review]

**Review by prashantkawa — APPROVED** [review]
Approving PR

**Pull request merged**